cpp/src/arrow/dataset/file_parquet_encryption_test.cc

```diff
@@ -51,8 +53,15 @@ using arrow::internal::checked_pointer_cast;
 namespace arrow {
 namespace dataset {
 
+// Tests come in these variations
+enum EncryptionParam {
+  COLUMN_KEY,
+  UNIFORM,
+};
+
 // Base class to test writing and reading encrypted dataset.
-class DatasetEncryptionTestBase : public ::testing::Test {
+class DatasetEncryptionTestBase
+    : public testing::TestWithParam<std::tuple<EncryptionParam, bool>> {
```

Comment:
I think this would be more readable with something like:
```c++
struct EncryptionTestParam {
  bool uniform_encryption;   // false is using per-column keys
  bool parallel;
};

class DatasetEncryptionTestBase : public testing::TestWithParam<EncryptionTestParam> {
```